sdk/apimanagement/api-management-custom-widgets-tools/src/node/CustomWidgetBlobService.ts

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { BlobServiceClient, BlockBlobUploadResponse } from "@azure/storage-blob";
+import { buildBlobConfigSrc, buildBlobDataSrc } from "../paths";
+import mime from "mime";
+
+type TConfig = Record<string, unknown>;
+
+/**
+ * A service wrapping ContainerClient class to simplify blob handling
+ */
+class CustomWidgetBlobService {
+  containerClient;
+  name;
+  pathWidget;
+  pathConfig;
+
+  /**
+   * @param blobStorageUrl - blob storage SAS URL
+   * @param name - name of the custom widget to be taken care of
+   */
+  constructor(blobStorageUrl: string, name: string) {
+    const container = "content";
+    const blobServiceClient = new BlobServiceClient(blobStorageUrl.replace(`/${container}`, ""));
+    this.containerClient = blobServiceClient.getContainerClient(container);
+    this.name = name;
+    this.pathWidget = buildBlobDataSrc(name);
+    this.pathConfig = buildBlobConfigSrc(name);
+  }
+
+  fileName(path: string): string | undefined {
```

Comment:
Should all of these functions be public? What about naming it `getFileName`, `extractFileName` or `fileNameFromPath`. Let's make all helper functions private